src/microbe_stage/systems/MicrobeProcessManagerSystem.cs

```diff
@@ -0,0 +1,180 @@
+        {
+        }
+
+        /// <summary>
+        ///   Sets the biome whose environmental values affect processes
+        /// </summary>
+        public void SetBiome(BiomeConditions newBiome)
+        {
+            biome = newBiome;
+        }
+
+        protected override void Update(float delta, in Entity entity)
+        {
+            ref var storage = ref entity.Get<CompoundStorage>();
+            ref var processes = ref entity.Get<BioProcesses>();
+
+            ManageProcesses(ref processes, ref storage, delta);
+        }
+
+        private float GetPriority(BioProcess process)
+        {
+            var key = process;
+
+            if (cachedPriority.TryGetValue(key, out var cached))
+            {
+                return cached;
+            }
+
+            // Lower priorities get to run first, so start at bottom
+            // It is important non-ATP-producers run first
+            // Different levels of priority "coarseness" are stored in different place values
+            var priority = 0.0f;
+
+            var inputAmount = 0.0f;
+            foreach (var entry in process.Inputs)
+            {
+                if (entry.Key.IsEnvironmental)
+                    continue;
+
+                inputAmount += entry.Value;
+            }
+
+            if (process.Outputs.TryGetValue(ATP, out var atpOut))
+            {
+                if (inputAmount <= 0)
+                {
+                    priority = 0;
+                }
+                else
+                {
+                    priority = 10 - atpOut / inputAmount / 100;
+                }
+            }
+
+            // Penalize Glucose consumers to allow RuBisCo to work
+            if (process.Inputs.ContainsKey(Glucose))
+            {
+                priority += 100;
+            }
+
+            cachedPriority.Add(key, priority);
+            return priority;
+        }
+
+        private void ManageProcesses(ref BioProcesses processor, ref CompoundStorage storage, float delta)
+        {
+            // Make sure there are ActiveProcesses
+            if (processor.ActiveProcesses == null)
+                return;
+
+            if (biome == null)
+                throw new NullReferenceException("Biome needs to be set");
+
+            var glucoseConsuming = false;
+
+            // Create dummy compound bag
+            var bag = storage.Compounds.Clone();
+
+            // Sort processes in order of priority
+            // TODO: Possibly find a way to not run this on every call
+            var processPriority = processor.ActiveProcesses
+                .OrderBy(p => GetPriority(p.Process)).ToList();
+
+            processor.ActiveProcesses = processPriority;
+
+            foreach (var process in processor.ActiveProcesses)
+            {
+                process.Rate = 1;
+            }
+
+            foreach (var process in processor.ActiveProcesses)
```

Comment:
I really think there should be a way for the process system to store the information as it is calculating it for this system to use. Now this basically doubles the amount of calculation needed to run the process system and makes the overall thing *much* harder to change in the future.

Reply:
Could you elaborate exactly what you mean by this? I now see a way to remove the storageConstraintModifier from this process which would be the main culprit here. I think ProcessSystem.CalculateEnvironmentModifier would heavily benefit from some sort of caching, but I'm not sure the best way to go about it.

Reply:
I'd start by modifying the `BioProcesses` component. To start off I'd put in fields like "bool UsedGlucose" and `float AtpBalance`. Those fields would be written by the process system as it does its thing as it basically already needs to know that information it would be just a tiny bit of extra overhead to store that information. 
Based on those I'd write a control algorithm where the speed for rubisco was gradually increased as long as UsedGlucose was false and AtpBalance was over 0.5f or something. Then conversely if I just didn't icrease the rubisco speed, I'd maybe half it? Or at least reduce it a lot more than I increment it. (halfing would be very similar to how TCP flow control adjusts the speed).